File: app/src/main/java/tw/y_studio/ptt/Utils/UIUtils.java
package tw.y_studio.ptt.Utils;

import android.content.Context;
import android.content.Intent;

import org.jetbrains.annotations.NotNull;

public class UIUtils {


    /**
     * 使用Android分享選單分享
     * @param context
     * @param subject
     * @param body
     * @param chooserTitle
     */
    public static void shareTo(@NotNull Context context, String subject, String body, String chooserTitle) {
        Intent sharingIntent = new Intent(android.content.Intent.ACTION_SEND);
        sharingIntent.setType("text/plain");
        sharingIntent.putExtra(android.content.Intent.EXTRA_SUBJECT, subject);
        sharingIntent.putExtra(android.content.Intent.EXTRA_TEXT, body);
        context.startActivity(Intent.createChooser(sharingIntent, chooserTitle));
    }
}

File: app/src/main/java/tw/y_studio/ptt/Utils/UIUtils.kt
@file:JvmName("UIUtils")

package tw.y_studio.ptt.Utils

import android.content.Context
import android.content.Intent

/**
 * 使用Android分享選單分享
 *
 * @param context
 * @param subject 主題
 * @param body 內文
 * @param chooserTitle 選擇器標題
 */
fun shareTo(context: Context, subject: String, body: String, chooserTitle: String) {
    val sharingIntent = Intent(Intent.ACTION_SEND).apply {
        type = "text/plain"
        putExtra(Intent.EXTRA_SUBJECT, subject)
        putExtra(Intent.EXTRA_TEXT, body)
    }
    context.startActivity(Intent.createChooser(sharingIntent, chooserTitle))
}
